src/ResourceManager/ApplicationInsights/Commands.ApplicationInsights/ApplicationInsights/SetApplicationInsightsPricingPlan.cs

```diff
@@ -0,0 +1,163 @@
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ApplicationInsights.Models;
+using Microsoft.Azure.Management.ApplicationInsights.Management.Models;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.ApplicationInsights
+{
+    [Cmdlet(VerbsCommon.Set, ApplicationInsightsPricingPlanNounStr), OutputType(typeof(PSPricingPlan))]
+    public class SetApplicationInsightsPricingPlanCommand : ApplicationInsightsBaseCmdlet
+    {
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ComponentObjectParameterSet,
+            ValueFromPipeline = true,
+            HelpMessage = "Application Insights Component Object.")]
+        [ValidateNotNull]
+        public PSApplicationInsightsComponent ApplicationInsightsComponent { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Application Insights Component Resource Id.")]
+        [ValidateNotNull]
+        public ResourceIdentifier ResourceId { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ComponentNameParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Resource Group Name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = ComponentNameParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Application Insights Component Name.")]
+        [Alias(ApplicationInsightsComponentNameAlias, ComponentNameAlias)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Position = 2,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Pricing plan name.")]
+        [ValidateSet(PricingPlans.Basic,
+            PricingPlans.Enterprise,
+            PricingPlans.LimitedBasic,
+            IgnoreCase = true)]
+        [ValidateNotNullOrEmpty]
+        public string PricingPlan { get; set; }
+
+        [Parameter(
+            Position = 3,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Daily Cap.")]        
+        public double? DailyCapGB { get; set; }
+
+        [Parameter(
+            Position = 4,
+            Mandatory = false,
+            HelpMessage = "Stop send notification when hit cap.")]        
+        public SwitchParameter DisableNotificationWhenHitCap { get; set; }
```

Comment:
@frankguodongchen same comment about `SwitchParameters` not having positions and making the enable and disable switches disjoint

Reply:
updated. thanks!